control/utils.py

```diff
@@ -182,6 +183,48 @@ def is_valid_nqn(nqn):
 
         return (0, os.strerror(0))
 
+class GatewayUtilsCrypto:
+    KEY_SIZE = 32
+
+    def __init__(self, encryption_key: bytes):
+        if encryption_key:
+            self.__secret_box = Fernet(encryption_key)
+        else:
+            self.__secret_box = None
+
+    @staticmethod
+    def read_encryption_key(keyfile : str) -> bytes:
+        keyval = ""
+        encoded_key = None
+        with open(keyfile) as f:
+            for line in f:
+                if line.startswith("-----BEGIN PRIVATE KEY-----"):
+                    continue
+                if line.startswith("-----END PRIVATE KEY-----"):
+                    continue
+                keyval += line.rstrip('\n')
+
+        keybytes = base64.b64decode(keyval)
+        if len(keybytes) < GatewayUtilsCrypto.KEY_SIZE:
+            raise RuntimeError(f"Encryption key has length {len(keybytes)} which is too short. The minimal length is {GatewayUtilsCrypto.KEY_SIZE}")
+        encoded_key = base64.urlsafe_b64encode(keybytes[:GatewayUtilsCrypto.KEY_SIZE])
```

Comment:
As long as you need something from the class this `@staticmethod` belongs to... is no longer a `@staticmethod`, but a `@classmethod`:
```suggestion
    @classmethod
    def read_encryption_key(cls, keyfile : str) -> bytes:
        keyval = ""
        encoded_key = None
        with open(keyfile) as f:
            for line in f:
                if line.startswith("-----BEGIN PRIVATE KEY-----"):
                    continue
                if line.startswith("-----END PRIVATE KEY-----"):
                    continue
                keyval += line.rstrip('\n')

        keybytes = base64.b64decode(keyval)
        if len(keybytes) < cls.KEY_SIZE:
            raise RuntimeError(f"Encryption key has length {len(keybytes)} which is too short. The minimal length is {cls.KEY_SIZE}")
        encoded_key = base64.urlsafe_b64encode(keybytes[:cls.KEY_SIZE])
```

Reply:
@epuertat I don't really need anything from the class. I just added the key length of 32 as a constant as I couldn't find it in the crypto package. I didn't think that this requires using a class method.

Reply:
Changed it to a class method